pkg/registry/service/rest.go

```diff
@@ -234,6 +234,16 @@ func (rs *REST) Delete(ctx api.Context, id string) (runtime.Object, error) {
 		}
 	}
 
+	if shouldCheckOrAssignHealthCheckNodePort(service) {
+		nodePort := apiservice.GetServiceHealthCheckNodePort(service)
+		if nodePort > 0 {
+			err := rs.serviceNodePorts.Release(int(nodePort))
+			if err != nil {
+				// these should be caught by an eventual reconciliation / restart
+				glog.Errorf("Error releasing service health check %s node port %d: %v", service.Name, nodePort, err)
```

Comment:
use `utilruntime.HandleError` anytime you eat an error (that allows us to plug in error shipping plugins and track them externally).  The default behavior is glog.Errorf so is exactly equal